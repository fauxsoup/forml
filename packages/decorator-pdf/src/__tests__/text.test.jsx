import Text from "../text";
import Context from "@forml/context";
import React from "react";
import ReactPDF from "@react-pdf/renderer";
import { render } from "@testing-library/react";
import * as decorator from "../";

describe("renders", function () {
  let styles;
  let text;
  let form = { type: "fieldset", items: [] };

  beforeEach(function () {
    text = jest.fn(() => ({ border: "1px solid red" }));
    styles = {
      get text() {
        return text();
      },
    };
  });

  test("without formStyles", function () {
    const { container } = render(
      <ReactPDF.Page>
        <Context.Provider value={{ decorator }}>
          <Text form={form} />
        </Context.Provider>
      </ReactPDF.Page>
    );

    expect(container).toMatchSnapshot();
  });
  test("with relevant formStyles", function () {
    form = { type: "help", styles, key: [], description: "description" };

    const { container } = render(
      <ReactPDF.Page>
        <Context.Provider value={{ decorator }}>
          <Text form={form} />
        </Context.Provider>
      </ReactPDF.Page>
    );

    expect(container).toMatchSnapshot();
    expect(text).toHaveBeenCalled();
  });
});
