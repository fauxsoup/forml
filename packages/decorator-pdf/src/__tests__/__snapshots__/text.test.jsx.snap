// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with relevant formStyles 1`] = `
<div>
  <page>
    <text
      style="font-size: 9pt; font-weight: 300; border: 1px solid red;"
    />
  </page>
</div>
`;

exports[`renders without formStyles 1`] = `
<div>
  <page>
    <text
      style="font-size: 9pt; font-weight: 300;"
    />
  </page>
</div>
`;
